cdap-explore/src/main/java/co/cask/cdap/explore/service/hive/BaseHiveExploreService.java

```diff
@@ -1049,20 +1050,24 @@ private void upgradeFilesetTable(TableInfo tableInfo) throws Exception {
     // enable the new table
     enableDataset(datasetID, spec);
 
-    Dataset dataset = ExploreServiceUtils.instantiateDataset(datasetFramework, datasetID);
-    // if this is a time partitioned file set, we need to add all partitions
-    if (dataset instanceof TimePartitionedFileSet) {
-      TimePartitionedFileSet tpfs = (TimePartitionedFileSet) dataset;
-      Set<PartitionDetail> partitionDetails = tpfs.getPartitions(null);
-      if (!partitionDetails.isEmpty()) {
-        QueryHandle handle = exploreTableManager.addPartitions(datasetID, partitionDetails);
-        QueryStatus status = waitForCompletion(handle);
-        // if add partitions failed, stop
-        if (status.getStatus() != QueryStatus.OpStatus.FINISHED) {
-          throw new ExploreException("Failed to add all partitions to dataset " + datasetID);
+    try (SystemDatasetInstantiator datasetInstantiator = datasetInstantiatorFactory.create()) {
+      Dataset dataset = datasetInstantiator.getDataset(datasetID);
+
+      // if this is a time partitioned file set, we need to add all partitions
+      if (dataset instanceof TimePartitionedFileSet) {
```

Comment:
this was upgrade code for the namespace release. I think PartitionedFileSets came after so there was no handling for that.  I'll add some comments